runtime/jvmti/jvmtiStackFrame.c

```diff
@@ -512,18 +512,27 @@ jvmtiNotifyFramePop(jvmtiEnv* env,
 		rc = getVMThread(currentThread, thread, &targetThread, TRUE, TRUE);
 		if (rc == JVMTI_ERROR_NONE) {
 			vm->internalVMFunctions->haltThreadForInspection(currentThread, targetThread);
-			if ((currentThread == targetThread) || (targetThread->publicFlags & J9_PUBLIC_FLAGS_HALT_THREAD_JAVA_SUSPEND))  {
-				J9StackWalkState walkState;
+			if ((NULL == targetThread) || (currentThread == targetThread) || (targetThread->publicFlags & J9_PUBLIC_FLAGS_HALT_THREAD_JAVA_SUSPEND))  {
+				J9StackWalkState walkState = {0};
+				j9object_t contObject = NULL;
+				if (NULL == targetThread) {
+					contObject = (j9object_t)J9VMJAVALANGVIRTUALTHREAD_CONT(currentThread, threadObject);
+				}
 
-				rc = findDecompileInfo(currentThread, targetThread, (UDATA)depth, &walkState);
+				rc = findDecompileInfo(
+						currentThread, targetThread,
+#if JAVA_SPEC_VERSION >= 19
+						contObject,
+#endif /* JAVA_SPEC_VERSION >= 19 */
+						(UDATA)depth, &walkState);
 				if (JVMTI_ERROR_NONE == rc) {
 					J9ROMMethod* romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(walkState.method);
 
 					if (romMethod->modifiers & J9AccNative) {
 						rc = JVMTI_ERROR_OPAQUE_FRAME;
 					} else {
 #ifdef J9VM_JIT_FULL_SPEED_DEBUG
-						if (walkState.jitInfo != NULL) {
+						if ((NULL != walkState.jitInfo) && (NULL != targetThread)) {
 							UDATA inlineDepth = (UDATA)walkState.userData2;
 							vm->jitConfig->jitFramePopNotificationAdded(currentThread, &walkState, inlineDepth);
```

Comment:
You can't just assume this only work for platform thread but not virtual thread

Yielded VirtualThread here won't work as `jitFramePopNotificationAdded` uses `walkState->walkThread` which will be out of scope after the `findDecompileInfo` call, we have to find a different option for this.